Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -287,34 +288,49 @@ def cherry_pick_pr!(user, repo, pr, args:, path: ".")
     Utils::Git.cherry_pick!(path, "--ff", "--allow-empty", *commits, verbose: args.verbose?, resolve: args.resolve?)
   end
 
-  def formulae_need_bottles?(tap, original_commit, user, repo, pr, args:)
+  def packages_need_bottles?(tap, original_commit, user, repo, pr, args:)
     return if args.dry_run?
 
     labels = GitHub.pull_request_labels(user, repo, pr)
 
     return false if labels.include?("CI-syntax-only") || labels.include?("CI-no-bottles")
 
-    changed_formulae(tap, original_commit).any? do |f|
-      !f.bottle_unneeded? && !f.bottle_disabled?
+    changed_packages(tap, original_commit).any? do |f|
+      !f.instance_of?(Cask::Cask) && !f.bottle_unneeded? && !f.bottle_disabled?
     end
   end
 
-  def changed_formulae(tap, original_commit)
+  def changed_packages(tap, original_commit)
     if Homebrew::EnvConfig.disable_load_formula?
       opoo "Can't check if updated bottles are necessary as HOMEBREW_DISABLE_LOAD_FORMULA is set!"
       return
     end
 
-    Utils.popen_read("git", "-C", tap.path, "diff-tree",
-                     "-r", "--name-only", "--diff-filter=AM",
-                     original_commit, "HEAD", "--", tap.formula_dir)
-         .lines
-         .map do |line|
+    formulae = Utils.popen_read("git", "-C", tap.path, "diff-tree",
+                                "-r", "--name-only", "--diff-filter=AM",
+                                original_commit, "HEAD", "--", tap.formula_dir)
+                    .lines
+                    .map do |line|
       next unless line.end_with? ".rb\n"
 
       name = "#{tap.name}/#{File.basename(line.chomp, ".rb")}"
-      Formula[name]
+      begin
+        Formulary.resolve(name)
+      rescue FormulaUnavailableError
+        nil
+      end
+    end.compact
+    casks = Utils.popen_read("git", "-C", tap.path, "diff-tree",
+                             "-r", "--name-only", "--diff-filter=AM",
+                             original_commit, "HEAD", "--", tap.cask_dir)
+                 .lines
+                 .map do |line|
+      next unless line.end_with? ".rb\n"
+
+      name = "#{tap.name}/#{File.basename(line.chomp, ".rb")}"
+      Cask::CaskLoader.load(name)
```

Comment:
Do we need to handle `CaskUnavailableError`s here like we do above for formulae?

Reply:
@SMillerDev nudge on this.

Reply:
Nudge on this.